terraform/provider/provider.go

```diff
@@ -37,42 +42,100 @@ func Provider() *schema.Provider {
 			},
 		},
 		ResourcesMap: map[string]*schema.Resource{
-			"teleport_user": resourceTeleportUser(),
 			"teleport_role": resourceTeleportRole(),
+			"teleport_user": resourceTeleportUser(),
 		},
-		ConfigureFunc: providerConfigure,
+		ConfigureContextFunc: configureContext,
 	}
 }
 
-// providerConfigure accepts provider settings schema,
+// configureContext accepts provider settings schema,
 // initializes a Teleport's `auth.Client` and returns it.
-func providerConfigure(d *schema.ResourceData) (interface{}, error) {
-	certPath := d.Get("cert_path").(string)
-	keyPath := d.Get("key_path").(string)
-	rootCAsPath := d.Get("root_ca_path").(string)
+func configureContext(ctx context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
+	certPath, err := getPath(d, "cert_path")
+	if err != nil {
+		return nil, diag.FromErr(trace.Wrap(err))
+	}
+
+	keyPath, err := getPath(d, "key_path")
+	if err != nil {
+		return nil, diag.FromErr(trace.Wrap(err))
+	}
 
-	// TODO: Verify Addr format is [protocol://]host:port
-	addr := d.Get("addr").(string)
+	rootCAsPath, err := getPath(d, "root_ca_path")
+	if err != nil {
+		return nil, diag.FromErr(trace.Wrap(err))
+	}
 
-	tlsConfig, err := loadTLSConfig(certPath, keyPath, rootCAsPath)
+	addr, err := getAddr(d, "addr")
 	if err != nil {
-		// TODO decorate the error into a Terraform-friendly Diags
-		return nil, trace.Wrap(err)
+		return nil, diag.FromErr(trace.Wrap(err))
 	}
 
+	creds := client.LoadKeyPair(certPath, keyPath, rootCAsPath)
```

Comment:
Nit: Minor thing, but you don't really need this extra variable, just do 

```go
Credentials: []client.Credentials{
  client.LoadKeyPair(certPath, keyPath, rootCAsPath),
}
```

below.

Reply:
👍